components/blocks/upcomingEvents.tsx

```diff
@@ -1,46 +1,33 @@
-"use client";
-
-import Image from "next/image";
-import { useEffect, useState } from "react";
 import type { Template } from "tinacms";
 import { tinaField } from "tinacms/dist/react";
+import { EventImageClient } from "../../app/components/event-image-client";
 import { useFormatDates } from "../../hooks/useFormatDates";
-import client from "../../tina/__generated__/client";
 import { CustomLink } from "../customLink";
 import { EventsRelativeBox } from "../events/eventsRelativeBox";
 import { EventTrimmed } from "../filter/events";
 import { PresenterList } from "../presenters/presenterList";
 
-export const UpcomingEvents = ({ data }) => {
-  const [events, setEvents] = useState<EventTrimmed[]>([]);
-  const [loading, setLoading] = useState(false);
+const fetchEvents = (data) => {
+  const events = data.events;
 
-  useEffect(() => {
-    const fetchEvents = async () => {
-      setLoading(true);
-      const today = new Date();
-      today.setHours(0, 0, 0, 0);
-      const events =
-        data.events ||
-        (await client.queries.getFutureEventsQuery({
-          fromDate: today.toISOString(),
-          top: data.numberOfEvents,
-        }));
-      setLoading(false);
+  if (!events) {
+    return [];
```

Comment:
I'd maybe add a check here to see if the page is in edit mode and populate the events using the graphql fetch if so.

I'm not seeing the events in the admin view on staging. (see docs for checking if Tina is in edit mode https://tina.io/docs/extending-tina/edit-state/)

![image](https://github.com/user-attachments/assets/1e71c908-b947-429c-9bce-b5492250355d)

Reply:
Fixed the issue with the client side rendering of upcoming events. 
![image](https://github.com/user-attachments/assets/98c56656-707f-4d90-aaa7-e8e96fb14bde)